AX: Import WPT accname tests.

AndresGonzalezApple opened the pull request:
#### 001bd0d42e302190f5936b03957b21f3e427fc82
<pre>
AX: Import WPT accname tests.
<a href="https://bugs.webkit.org/show_bug.cgi?id=275059">https://bugs.webkit.org/show_bug.cgi?id=275059</a>
&lt;<a href="https://rdar.apple.com/problem/129164464">rdar://problem/129164464</a>&gt;

Reviewed by Tyler Wilcock.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/a975c0fc84f2a43f7a9bccc3d8f1d68f3d348828">https://github.com/web-platform-tests/wpt/commit/a975c0fc84f2a43f7a9bccc3d8f1d68f3d348828</a>

In addition to importing the WPT tests as of today, this patch includes a fix for the method to determine whether we can determine the accessibility role and label of a given element based on the status of the RenderTree layout. The previous method caused the comp_label.html new test case to timeout because the button inside a span with display:none never has a RenderObject.

* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_label-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_label.html:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.html:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/accname/name/w3c-import.log:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/accname/name/comp_label-expected.txt:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::readyToRetrieveComputedRoleOrLabel const):

Canonical link: <a href="https://commits.webkit.org/279779@main">https://commits.webkit.org/279779@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/be094a8848fe9eac28208fa0c0bf0599c006b83f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54446 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33858 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7096 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57725 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5177 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56749 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41392 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5213 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44086 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3467 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56541 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/31999 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47223 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25222 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28801 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4553 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3320 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4767 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59316 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29669 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4888 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51511 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30824 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47310 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50881 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11914 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31802 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30617 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

twilco commented on LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_label-expected.txt:
```diff
@@ -245,36 +244,27 @@ PASS label valid on td element
+NOTRUN button with empty aria-label does not use aria-label as name
+NOTRUN textarea with tab character as aria-label does not use aria-label as name
+NOTRUN button with carriage return as aria-label does not use aria-label as name
+NOTRUN button with space characters as aria-label does not use aria-label as name
```
Updating all these `PASS` expectations to `NOTRUN` doesn't seem right.

AndresGonzalezApple committed:
AX: Import WPT accname tests.
https://bugs.webkit.org/show_bug.cgi?id=275059
<rdar://problem/129164464>

Reviewed by Tyler Wilcock.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/a975c0fc84f2a43f7a9bccc3d8f1d68f3d348828

In addition to importing the WPT tests as of today, this patch includes a fix for the method to determine whether we can determine the accessibility role and label of a given element based on the status of the RenderTree layout. The previous method caused the comp_label.html new test case to timeout because the button inside a span with display:none never has a RenderObject.

* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_label-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_label.html:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.html:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/accname/name/w3c-import.log:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/accname/name/comp_label-expected.txt:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::readyToRetrieveComputedRoleOrLabel const):

Canonical link: https://commits.webkit.org/279779@main
